### ISSUE TITLE
Apply error when ignore AMI in ec2

### ISSUE THREAD

**Issue by Meabed** [hashicorp/terraform#6887]
Hi,
### Terraform Version
    Terraform Version: 0.6.16

### Affected Resource(s)
- aws_instance

### Debug Output
https://gist.github.com/Meabed/181547598609960dd5b3768d34a07df0

### Expected Behavior
Should not show the error, when i apply it again it works without showing the error.

### Actual Behavior
It shows error that diff is wrong.

### Steps to Reproduce
1. `terraform apply`
2. Change AMI configuration and in the aws_instance add 
```
  lifecycle {
    ignore_changes = [
      "ami",
      "source_dest_check",
      "associate_public_ip_address",
      "instance_type",
      "root_block_device.0.delete_on_termination"
    ]
  }
```
2. `terraform apply`

**Comment by bmalynovytch** [comment]
sounds related to #6379

**Comment by stack72** [comment]
Hi @Meabed 

My apologies that we took so long to get back to you on this - there is no excuse for an issue to be open so long. I have just tested this as follows:

Config
```
provider "aws" {
  region = "us-west-2"
}

resource "aws_security_group" "tf_test_foo" {
	name = "tf_test_foo"
	description = "foo"

	ingress {
		protocol = "icmp"
		from_port = -1
		to_port = -1
		cidr_blocks = ["0.0.0.0/0"]
	}
}

resource "aws_instance" "foo" {
	ami = "ami-4fccb37f"
	availability_zone = "us-west-2a"

	instance_type = "m1.small"
	security_groups = ["${aws_security_group.tf_test_foo.name}"]

  lifecycle {
    ignore_changes = [
      "ami"
    ]
  }
}
```

Initial Plan and apply went fine:

```
% terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but
will not be persisted to local or remote state storage.


The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed. Cyan entries are data sources to be read.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

+ aws_instance.foo
    ami:                        "ami-4fccb37f"
    availability_zone:          "us-west-2a"
    ebs_block_device.#:         "<computed>"
    ephemeral_block_device.#:   "<computed>"
    instance_state:             "<computed>"
    instance_type:              "m1.small"
    key_name:                   "<computed>"
    network_interface_id:       "<computed>"
    placement_group:            "<computed>"
    private_dns:                "<computed>"
    private_ip:                 "<computed>"
    public_dns:                 "<computed>"
    public_ip:                  "<computed>"
    root_block_device.#:        "<computed>"
    security_groups.#:          "1"
    security_groups.3292662914: "tf_test_foo"
    source_dest_check:          "true"
    subnet_id:                  "<computed>"
    tenancy:                    "<computed>"
    vpc_security_group_ids.#:   "<computed>"

+ aws_security_group.tf_test_foo
    description:                          "foo"
    egress.#:                             "<computed>"
    ingress.#:                            "1"
    ingress.1799340084.cidr_blocks.#:     "1"
    ingress.1799340084.cidr_blocks.0:     "0.0.0.0/0"
    ingress.1799340084.from_port:         "-1"
    ingress.1799340084.protocol:          "icmp"
    ingress.1799340084.security_groups.#: "0"
    ingress.1799340084.self:              "false"
    ingress.1799340084.to_port:           "-1"
    name:                                 "tf_test_foo"
    owner_id:                             "<computed>"
    vpc_id:                               "<computed>"


Plan: 2 to add, 0 to change, 0 to destroy.
```
```
[stacko@Pauls-MacBook-Pro:~/Code/terraform-recreations/6887]
% terraform apply
aws_security_group.tf_test_foo: Creating...
  description:                          "" => "foo"
  egress.#:                             "" => "<computed>"
  ingress.#:                            "" => "1"
  ingress.1799340084.cidr_blocks.#:     "" => "1"
  ingress.1799340084.cidr_blocks.0:     "" => "0.0.0.0/0"
  ingress.1799340084.from_port:         "" => "-1"
  ingress.1799340084.protocol:          "" => "icmp"
  ingress.1799340084.security_groups.#: "" => "0"
  ingress.1799340084.self:              "" => "false"
  ingress.1799340084.to_port:           "" => "-1"
  name:                                 "" => "tf_test_foo"
  owner_id:                             "" => "<computed>"
  vpc_id:                               "" => "<computed>"
aws_security_group.tf_test_foo: Creation complete
aws_instance.foo: Creating...
  ami:                        "" => "ami-4fccb37f"
  availability_zone:          "" => "us-west-2a"
  ebs_block_device.#:         "" => "<computed>"
  ephemeral_block_device.#:   "" => "<computed>"
  instance_state:             "" => "<computed>"
  instance_type:              "" => "m1.small"
  key_name:                   "" => "<computed>"
  network_interface_id:       "" => "<computed>"
  placement_group:            "" => "<computed>"
  private_dns:                "" => "<computed>"
  private_ip:                 "" => "<computed>"
  public_dns:                 "" => "<computed>"
  public_ip:                  "" => "<computed>"
  root_block_device.#:        "" => "<computed>"
  security_groups.#:          "" => "1"
  security_groups.3292662914: "" => "tf_test_foo"
  source_dest_check:          "" => "true"
  subnet_id:                  "" => "<computed>"
  tenancy:                    "" => "<computed>"
  vpc_security_group_ids.#:   "" => "<computed>"
aws_instance.foo: Still creating... (10s elapsed)
aws_instance.foo: Still creating... (20s elapsed)
aws_instance.foo: Still creating... (30s elapsed)
aws_instance.foo: Creation complete

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: terraform.tfstate
```

I think changed the AMI to `ami-797ea219`

I ran another plan and got the following:

```
% terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but
will not be persisted to local or remote state storage.

aws_security_group.tf_test_foo: Refreshing state... (ID: sg-978325ee)
aws_instance.foo: Refreshing state... (ID: i-01516c19)

No changes. Infrastructure is up-to-date. This means that Terraform
could not detect any differences between your configuration and
the real physical resources that exist. As a result, Terraform
doesn't need to do anything.
```

An apply looks as follows:

```
% terraform apply
aws_security_group.tf_test_foo: Refreshing state... (ID: sg-978325ee)
aws_instance.foo: Refreshing state... (ID: i-01516c19)

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

This issue has been fixed in the work completed for 0.7 release!

Hope this is ok 

Paul

**Comment by Meabed** [comment]
Thank you, its working fine on the last release 👍
